source/postprocess/visualization/heating.cc

```diff
@@ -120,17 +125,31 @@ namespace aspect
 
             for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
               in.composition[q][c] = uh[q][this->introspection().component_indices.compositional_fields[c]];
+
+            mid_point += evaluation_points[q]/n_quadrature_points;
           }
 
         in.position = evaluation_points;
 
-        this->get_material_model().evaluate(in, out);
+        typename DoFHandler<dim>::active_cell_iterator cell;
+        cell = (GridTools::find_active_cell_around_point<> (this->get_mapping(), this->get_dof_handler(), mid_point)).first;
```

Comment:
No suggestions. This is blocked by https://github.com/dealii/dealii/issues/391 unfortunately.

If you merge this, please open a github issue that reminds us that we fix this after https://github.com/dealii/dealii/issues/391 is fixed.